lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java

```diff
@@ -907,6 +907,24 @@ void testRenameFile()
                 }
             }
         }
+        finally {
+            // clean up manually created parent directory
+            getFileSystem().deleteFile(createLocation("renameTarget"));
+        }
+
+        // rename to a file with special characters in the name
+        try (TempBlob sourceBlob = randomBlobLocation("renameSource");
+                TempBlob targetBlob = randomBlobLocation("renameTarget%25special")) {
+            sourceBlob.createOrOverwrite("data");
+            getFileSystem().renameFile(sourceBlob.location(), targetBlob.location());
+            assertThat(sourceBlob.exists()).isFalse();
+            assertThat(targetBlob.exists()).isTrue();
+            assertThat(targetBlob.read()).isEqualTo("data");
+        }
+        finally {
+            // clean up manually created parent directory
+            getFileSystem().deleteFile(createLocation("renameTarget%25special"));
```

Comment:
Seems that we create blob with random suffix and here we try to delete location without that suffix.

Reply:
Yeah, I have no idea why this is needed. We create a new empty directory, which actually also creates a file, that needs to be manually removed afterward. This is a preexisting condition, but I can't explain why this wasn't causing test issues before. Some tests make sure that locations are empty before running. Maybe tests order changed.